File: impls/ts/step0_repl.ts
const readline = require("readline");

const rep = (input: string): string => {
  return input;
};

process.stdout.write("user> ");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
  terminal: true
});
rl.on("line", (line: string) => {
  console.log(rep(line));
  process.stdout.write("user> ");
});

File: impls/rust/step0_repl.rs
use rustyline::Editor;

fn main() {
    let mut rl = Editor::<()>::new();
    loop {
        let readline = rl.readline("user> ");
        match readline {
            Ok(line) => {
                println!("{}", line);
            },
            Err(_) => {
                break
            },
        }
    }
}
